AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -1683,6 +1683,18 @@ - (void)recursivelyClearFetchedData
   [self clearFetchedData];
 }
 
+- (ASInterfaceState)interfaceState
+{
+  return _interfaceState;
+}
+
+- (void)setInterfaceState:(ASInterfaceState)interfaceState
+{
+  if (interfaceState != _interfaceState) {
```

Comment:
Added the locking, but decided to wait on interfaceStateDidChange: until we know what the protocol + adoption with ASViewController / table / collection will look like.  The main known use case involves simply reading self.interfaceState (which is accessible) on setURL: or similar to check if a load should be kicked off.  It is also possible to override -setInterfaceState: and call super if needed, but I agree this is not ideal for long term usage outside the framework.